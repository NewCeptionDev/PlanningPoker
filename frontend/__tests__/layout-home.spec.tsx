import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Layout from 'src/app/layout';

describe('layout', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  })

  it('should render layout unchanged', () => {
    const { container } = render(<Layout />);
    expect(container).toMatchSnapshot();
  });

  it('should render links', () => {
    render(<Layout />);

    expect(screen.getByTestId('favicon-32')).toBeInTheDocument();
    expect(screen.getByTestId('favicon-16')).toBeInTheDocument();
    expect(screen.getByTestId('favicon-apple-16')).toBeInTheDocument();
    expect(screen.getByTestId('favicon-apple-32')).toBeInTheDocument();
  });
});
