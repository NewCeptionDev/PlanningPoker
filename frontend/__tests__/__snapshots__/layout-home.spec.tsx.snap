// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should render layout unchanged 1`] = `
<div>
  <html
    lang="en"
  >
    <link
      data-testid="favicon-32"
      href="/favicon-32x32.png"
      rel="icon"
      sizes="32x32"
    />
    <link
      data-testid="favicon-16"
      href="/favicon-16x16.png"
      rel="icon"
      sizes="16x16"
    />
    <link
      data-testid="favicon-apple-16"
      href="/favicon-16x16.png"
      rel="apple-touch-icon"
      sizes="16x16"
    />
    <link
      data-testid="favicon-apple-32"
      href="/favicon-32x32.png"
      rel="apple-touch-icon"
      sizes="32x32"
    />
    <body
      class="className"
      data-testid="body"
    >
      <script>
        !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
      </script>
    </body>
  </html>
</div>
`;
